### PR TITLE
Allow disabling health extension with configuration

### LINKED ISSUE #42356: Don't expose health probes with configuration
### Description

Originally requested here https://github.com/quarkiverse/quarkus-operator-sdk/issues/818#issuecomment-1942199819.

I think it would make sense. I can see similar configs in a few different extensions.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by xstefank**
Fixes #42356

### CODE REVIEW

**Comment by xstefank** [comment]
Also, it will probably supersede #38623 if @metacosm agrees.

File: extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/HealthBuildTimeConfig.java [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 @ConfigRoot(name = "health")
 public class HealthBuildTimeConfig {
+    /**
+     * Activate or disable this extension. Disabling this extension means that no health related information is exposed.
+     */
+    @ConfigItem(name = "active", defaultValue = "true")
+    public boolean active;
```
Comment (gsmet) [inline extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/HealthBuildTimeConfig.java:12]: If you disable the extension entirely, please use `enabled`, not `active`.

Also don't define a `name`, the convention will make it right.

See this issue for the overall context: https://github.com/quarkusio/quarkus/issues/42244

**Review by gsmet — CHANGES_REQUESTED** [review]
Please see my comment inline.

**Comment by xstefank** [comment]
@gsmet fixed. I also aligned that it will always produce the feature so it will be displayed on startup to align with hibernate.

**Review by gsmet — APPROVED** [review]
Perfect, thanks.

I will merge it as soon as CI is green as I need to make changes in this class.

**Pull request merged**